File: React/src/components/model/API/UserForRegistration.ts
import IUserForRegistration from "../../../interfaces/API/IUserForRegistration";

export class UserForRegistration implements IUserForRegistration {
    FirstName?: string;
    LastName?: string;
    UserName?: string;
    Email?: string;
    Password?: string;

    constructor(firstName?: string, lastName?: string, userName?: string, email?: string, password?: string, ) {
        this.FirstName = firstName;
        this.LastName = lastName;
        this.UserName = userName;
        this.Email = email;
        this.Password = password;
    }
}

File: Petroliq/Model/ControllerModels/UserForRegistration.cs
namespace Petroliq_API.Model.ControllerModels
{
    /// <summary>
    /// Intermediary User object specifically for Registration, doesn't include Id fields
    /// </summary>
    public class UserForRegistration
    {
#pragma warning disable CS1591
        public string? FirstName { get; set; }
        public string? LastName { get; set; } = string.Empty;
        public string? UserName { get; set; }
        public string? Email { get; set; }
        public string? Password { get; set; }
#pragma warning restore CS1591
    }
}
